datumaro/plugins/common_segmentation_format.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import glob
+import os.path as osp
+
+import numpy as np
+
+from datumaro.components.annotation import AnnotationType, LabelCategories, Mask, MaskCategories
+from datumaro.components.extractor import DatasetItem, Importer, SourceExtractor
+from datumaro.components.format_detection import FormatDetectionContext
+from datumaro.components.media import Image
+from datumaro.util.image import find_images
+from datumaro.util.mask_tools import generate_colormap, lazy_mask
+from datumaro.util.meta_file_util import (
+    DATASET_META_FILE,
+    has_meta_file,
+    is_meta_file,
+    parse_meta_file,
+)
+
+
+class CommonSegmentationPath:
+    MASKS_DIR = "masks"
+    IMAGES_DIR = "images"
+    IMAGE_EXT = ".png"
+
+
+def make_categories(label_map=None):
+    if label_map is None:
+        return {}
+
+    categories = {}
+    label_categories = LabelCategories()
+    for label in label_map:
+        label_categories.add(label)
+    categories[AnnotationType.label] = label_categories
+
+    has_colors = any(v is not None for v in label_map.values())
+    if not has_colors:  # generate new colors
+        colormap = generate_colormap(len(label_map))
+    else:  # only copy defined colors
+        label_id = lambda label: label_categories.find(label)[0]
+        colormap = {label_id(name): (desc[0], desc[1], desc[2]) for name, desc in label_map.items()}
+    mask_categories = MaskCategories(colormap)
+    mask_categories.inverse_colormap  # pylint: disable=pointless-statement
+    categories[AnnotationType.mask] = mask_categories
+    return categories
+
+
+class CommonSegmentationExtractor(SourceExtractor):
```

Comment:
I decided that the name `CommonSemanticSegmentation` is too long. 

We can use the CSS name, but since the format is not very well known, it will be difficult for most users to understand what it is for by this name.

Reply:
Given that in AC this format is referred to as `common_semantic_segmentation`, I think we should do the same.

Reply:
Renamed to `common_semantic_segmentation`